File: nodeProject/Models/Dog.js
const mongoose = require('mongoose');

const dogSchema = new mongoose.Schema({
    name: { type: String, required: true },
    age: { type: Number, required: true },
    gender: { type: String, required: true },
    color: { type: String, required: true },
    weight: { type: Number, required: true },
    distance: { type: String, required: true },
    imagePath: { type: String, required: true },
    description: { type: String }
});

const Dog = mongoose.model('Dog', dogSchema);

module.exports = Dog;

File: projectadopet/lib/models/dog.dart
class Dog {
  final String id;
  final String name;
  final int age;
  final String gender;
  final String color;
  final double weight;
  final String distance;
  final String description;
  final String imagePath;

  Dog({
    required this.id,
    required this.name,
    required this.age,
    required this.gender,
    required this.color,
    required this.weight,
    required this.distance,
    required this.description,
    required this.imagePath
  });

  // Convertir les données du chien en JSON
  Map<String, dynamic> toJson() {
    return {
      '_id': id, 
      'name': name,
      'age': age,
      'gender': gender,
      'color': color,
      'weight': weight,
      'distance': distance,
      'description': description,
      'imagePath': imagePath
    };
  }

  // Créer un objet Dog depuis un JSON
  factory Dog.fromJson(Map<String, dynamic> json) {
    return Dog(
      id: json['_id'],
      name: json['name'] ?? '',
      age: json['age'] ?? 0,
      gender: json['gender'] ?? '',
      color: json['color'] ?? '',
      weight: json['weight'] ?? 0.0,
      distance: json['distance'] ?? '',
      description: json['description'] ?? '',
      imagePath: json['imagePath'] ?? ''
    );
  }
}
